app/src/pages/settings/DBUsagePieChart.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react";
+import { graphql, useFragment } from "react-relay";
+import { schemePaired } from "d3-scale-chromatic";
+import {
+  Cell,
+  Pie,
+  PieChart,
+  ResponsiveContainer,
+  Tooltip,
+  TooltipProps,
+} from "recharts";
+
+import { ChartTooltip, ChartTooltipItem } from "@phoenix/components/chart";
+import { storageSizeFomatter } from "@phoenix/utils/storageSizeFormatUtils";
+
+import { DBUsagePieChart_data$key } from "./__generated__/DBUsagePieChart_data.graphql";
+
+function TooltipContent({ active, payload }: TooltipProps<number, string>) {
+  if (active && payload && payload.length) {
+    return (
+      <ChartTooltip>
+        <ChartTooltipItem
+          shape="square"
+          color={payload[0].payload.fill || "transparent"}
+          name={payload[0].name || "--"}
+          value={storageSizeFomatter(payload[0].value || 0)}
+        />
+      </ChartTooltip>
+    );
+  }
+
+  return null;
+}
+
+export function DBUsagePieChart({
+  query,
+}: {
+  query: DBUsagePieChart_data$key;
+}) {
+  const data = useFragment<DBUsagePieChart_data$key>(
+    graphql`
+      fragment DBUsagePieChart_data on Query {
+        dbTableStats {
+          tableName
+          numBytes
+        }
+      }
+    `,
+    query
+  );
+
+  const totalBytes = data.dbTableStats.reduce(
+    (acc, table) => acc + table.numBytes,
+    0
+  );
+
+  return (
+    <ResponsiveContainer width="100%" height={245}>
+      <PieChart>
+        <Pie
+          data={[...data.dbTableStats]}
+          dataKey="numBytes"
+          nameKey="tableName"
+          cx="50%"
+          cy="50%"
+          innerRadius={70}
+          outerRadius={110}
+          strokeWidth={0}
+          stroke="transparent"
+        >
+          {data.dbTableStats.map((x, index) => (
+            <Cell
+              stroke="0"
+              key={`cell-${index}`}
+              fill={`${schemePaired[index % schemePaired.length]}`}
+            />
+          ))}
+        </Pie>
+        <Tooltip content={<TooltipContent />} />
+        <g>
+          <text
+            x="50%"
+            y="50%"
+            dy={8}
+            textAnchor="middle"
+            fill="var(--ac-global-text-color-900"
+            fontSize="var(--ac-global-font-size-l)"
+          >
+            {storageSizeFomatter(totalBytes)}
```

Comment:
It's kinda hard to tell if this is space remaining or space used. Can we make this a ratio or add a label?

Reply:
I think it's not possible right now to know the total capacity. I'll add a label